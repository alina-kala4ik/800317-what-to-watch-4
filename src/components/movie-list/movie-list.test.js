import React from "react";
import renderer from "react-test-renderer";
import MovieList from "./movie-list.jsx";

const films = [
  {
    id: 1,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    src: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  },
  {
    id: 2,
    title: `Bohemian Rhapsody`,
    src: `img/bohemian-rhapsody.jpg`,
  },
  {
    id: 3,
    title: `Macbeth`,
    src: `img/macbeth.jpg`
  },
  {
    id: 4,
    title: `Aviator`,
    src: `img/aviator.jpg`,
  },
  {
    id: 5,
    title: `We need to talk about Kevin`,
    src: `img/we-need-to-talk-about-kevin.jpg`,
  },
  {
    id: 6,
    title: `What We Do in the Shadows`,
    src: `img/what-we-do-in-the-shadows.jpg`,
  },
  {
    id: 7,
    title: `Revenant`,
    src: `img/revenant.jpg`,
  },
  {
    id: 8,
    title: `Johnny English`,
    src: `img/johnny-english.jpg`,
  },
];

it(`render MovieList`, () => {
  const tree = renderer
    .create(<MovieList
      films={films}
      onFilmTitleClick={()=>{}}
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
