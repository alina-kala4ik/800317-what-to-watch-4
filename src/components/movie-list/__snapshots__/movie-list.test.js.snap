// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MovieList 1`] = `
Array [
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        autoPlay={false}
        height="175"
        loop={true}
        muted="muted"
        poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
        preload="none"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        autoPlay={false}
        height="175"
        loop={true}
        muted="muted"
        poster="img/bohemian-rhapsody.jpg"
        preload="none"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Bohemian Rhapsody
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        autoPlay={false}
        height="175"
        loop={true}
        muted="muted"
        poster="img/macbeth.jpg"
        preload="none"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Macbeth
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        autoPlay={false}
        height="175"
        loop={true}
        muted="muted"
        poster="img/aviator.jpg"
        preload="none"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Aviator
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        autoPlay={false}
        height="175"
        loop={true}
        muted="muted"
        poster="img/we-need-to-talk-about-kevin.jpg"
        preload="none"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        We need to talk about Kevin
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        autoPlay={false}
        height="175"
        loop={true}
        muted="muted"
        poster="img/what-we-do-in-the-shadows.jpg"
        preload="none"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        What We Do in the Shadows
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        autoPlay={false}
        height="175"
        loop={true}
        muted="muted"
        poster="img/revenant.jpg"
        preload="none"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Revenant
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
      onClick={[Function]}
    >
      <video
        autoPlay={false}
        height="175"
        loop={true}
        muted="muted"
        poster="img/johnny-english.jpg"
        preload="none"
        src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="movie-page.html"
        onClick={[Function]}
      >
        Johnny English
      </a>
    </h3>
  </article>,
]
`;
